src/policy/marksweepspace/native_ms/global.rs

```diff
@@ -341,15 +351,17 @@ impl<VM: VMBinding> MarkSweepSpace<VM> {
 
     pub fn release(&mut self) {
         if cfg!(feature = "eager_sweeping") {
-            // For eager sweeping, we have to sweep the lists that are abandoned to these global lists.
-            let mut abandoned = self.abandoned.lock().unwrap();
-            abandoned.sweep(self);
-        } else {
-            // For lazy sweeping, we just move blocks from consumed to unswept. When an allocator tries
-            // to use them, they will sweep the block.
-            let mut abandoned = self.abandoned.lock().unwrap();
-            abandoned.sweep_later(self);
+            let num_mutators = VM::VMActivePlan::number_of_mutators();
+            // all ReleaseMutator work packets plus the ReleaseMarkSweepSpace packet
+            self.pending_release_packets
```

Comment:
I suggest just using this mechanism for both eager and lazy sweeping. We always maintain the pending packets count. Just for eager sweeping, when all the packets are done, you generate sweep tasks. This could make things cleaner -- performance wise, the impact should be minimal. Otherwise, we see the conditionally compiled code for eager sweeping scattered everywhere to achieve this one thing.

Reply:
I moved what's in `MarkSweepSpace::end_of_gc` into `recycle_blocks`.  That is now used by both lazy and eager sweeping.  When using lazy sweeping, we directly call `recycle_blocks` after all `ReleaseMutator` packets and the `ReleaseMarkSweepSpace` packet have finished.